Add Bartlett's and Levene's tests for homogeneity of variances

PraneyaKumar opened the pull request:
## Summary
- Adds Bartlett's and Levene's tests for homogeneity of variances.
- Resolves #137.

## Changes
- **New Modules**:
  - `Statistics.Test.Bartlett`
  - `Statistics.Test.Levene`
- Added these modules to the `exposed-modules` section in the `statistics.cabal` file.
- **Documentation**:
  - Usage examples in Haddock comments.
- Updated the `changelog.md` to reflect the addition.

## Notes for Reviewers
- This is my first contribution to the library. Feedback on:
  - Numerical stability improvements
  - Error handling patterns
  - Test coverage
  - Any improvements needed in the code structure or style

Shimuuar commented on Statistics/Test/Bartlett.hs:
```diff
@@ -0,0 +1,92 @@
+    -- Degrees of freedom and chi-squared distribution
+    df = k - 1
+    chiDist = chiSquared df
+    pValue = mkPValue $ 1 - cumulative chiDist tStatistic
```
`complCumulative` should calculates 1-CDF without loss of precision

AFAIR χ² doesn't implement it but it may at some point

Shimuuar commented on Statistics/Test/Levene.hs:
```diff
@@ -0,0 +1,144 @@
+          zbari = map Sample.mean deviations
+          zbar = sum (zipWith (*) zbari (map fromIntegral ni)) / fromIntegral n
+
+          numerator = sum [fromIntegral (ni !! i) * (zbari !! i - zbar) ** 2 | i <- [0..k-1]]
```
`!!` is generally smell since it's O(n). `zbari` and `ni` should probably be vector

Shimuuar reviewed (commented):
Thank you for PR! I'll be busy whole April and will need to read about both tests. So thing will probably move slowly. 

There're few comments on things that caught my eye,

PraneyaKumar commented:
Hi @Shimuuar!

Thanks for the helpful feedback!

1. **`complCumulative`** – I noticed that `chiSquared` currently doesn’t implement `complCumulative`. So for now, I’ve added a `TODO` comment in the code noting that this could be replaced with `complCumulative` when it becomes available. 

2. **Avoiding `!!`** – I've refactored the `levenesTest` implementation to use boxed `Vector`s (`Data.Vector`) and replaced the list-indexing logic with `Vector.zipWith` and `Vector.sum`. This should be more idiomatic and efficient. Let me know if any further improvements are needed on this part.

Thanks again, and no worries at all about things moving slowly — I really appreciate your time and review whenever you get a chance.

Shimuuar commented on Statistics/Test/Levene.hs:
```diff
@@ -0,0 +1,149 @@
+levenesTest :: Double       -- ^ Significance level (alpha)
+            -> Center       -- ^ Center calculation method
+            -> [V.Vector Double] -- ^ Input samples
+            -> Either String TestResult
```
Function should return `Statistics.Test.Types.Test` as Bartlett do,

PraneyaKumar replied:
I have now reworked Levene's Test to return Test instead of TestResult.

Shimuuar reviewed (changes requested):
In general PR is fine.

> complCumulative – I noticed that chiSquared currently doesn’t implement complCumulative. So for now, I’ve added a TODO comment in the code noting that this could be replaced with complCumulative when it becomes available.

It's not implemented so default implementation will be used instead. And if class method gets implemented it will be used automatically, no need to change anything in test implementation.


----

Please also add simple unit test in `Tests.Parametric` that checks both test compute approximately same test statistics and p-value as python/R/whatever.

PraneyaKumar commented:
> In general PR is fine.
> 
> > complCumulative – I noticed that chiSquared currently doesn’t implement complCumulative. So for now, I’ve added a TODO comment in the code noting that this could be replaced with complCumulative when it becomes available.
> 
> It's not implemented so default implementation will be used instead. And if class method gets implemented it will be used automatically, no need to change anything in test implementation.
> 
> Please also add simple unit test in `Tests.Parametric` that checks both test compute approximately same test statistics and p-value as python/R/whatever.

Hi @Shimuuar!

I've pushed the requested changes:
- Reworked Levene's Test to return Test instead of TestResult.
- Added unit test for Bartlett's and Levene's test in Tests.Parametric, verifying the test statistic and p-value against reference values.

Please let me know if any further adjustments are needed!

Shimuuar commented:
Well. It needs quite a bit of improvements but it would be easier to just merge it as is and fix it later
